scripts/intfutil

```diff
@@ -506,10 +516,77 @@ class IntfDescription(object):
         if self.appl_db_keys:
             self.table += self.generate_intf_description()
 
+
+# ========================== interface-autoneg logic ==========================
+header_autoneg = ['Interface', 'Auto-Neg Mode', 'Speed', 'Adv Speeds', 'Type', 'Adv Types', 'Oper', 'Admin']
+
+
+class IntfAutoNegStatus(object):
+
+    def __init__(self, intf_name, namespace_option, display_option):
+        self.db = None
+        self.config_db = None
+        self.table = []
+        self.multi_asic = multi_asic_util.MultiAsic(
+            display_option, namespace_option)
+
+        if intf_name is not None and intf_name == SUB_PORT:
+            self.intf_name = None
+        else:
+            self.intf_name = intf_name
+
+    def display_autoneg_status(self):
+        
+        self.get_intf_autoneg_status()
+
+        # Sorting and tabulating the result table.
+        sorted_table = natsorted(self.table)
+        print(tabulate(sorted_table, header_autoneg, tablefmt="simple", stralign='right'))
+
+    def generate_autoneg_status(self):
+        """
+            Generate interface-autoneg output
+        """
+
+        i = {}
+        table = []
+        key = []
+
+        #
+        # Iterate through all the keys and append port's associated state to
+        # the result table.
+        #
+        for i in self.appl_db_keys:
+            key = re.split(':', i, maxsplit=1)[-1].strip()
+            if key in self.front_panel_ports_list:
+                if self.multi_asic.skip_display(constants.PORT_OBJ, key):
+                        continue
+                autoneg_mode = appl_db_port_status_get(self.db, key, PORT_AUTONEG)
+                if autoneg_mode != 'N/A':
+                    autoneg_mode = 'enabled' if autoneg_mode == 'on' else 'disabled'
+                table.append((key,
+                              autoneg_mode,
+                              appl_db_port_status_get(self.db, key, PORT_SPEED),
```

Comment:
I was thinking to fetch each field for each port from the database is not a good way. Probably we can fetch the entire table and then fetch info from the dict.

Reply:
We cannot get the entire table for the APPL_DB.